prisma/migrations/20250224065512_session_trigger/migration.sql

```diff
@@ -18,9 +18,52 @@ SET search_path = public
 AS $$
 DECLARE
     country_code VARCHAR(2);
+    user_exists BOOLEAN;
 BEGIN
-    INSERT INTO "Sessions" (id, "createdAt", "updatedAt", ip, "userAgent", "userId", "deviceNonce")
-    SELECT NEW.id, NEW.created_at, NEW.updated_at, NEW.ip, NEW.user_agent, NEW.user_id, gen_random_uuid();
+    -- Check if the user exists in UserProfiles
+    SELECT EXISTS (
+        SELECT 1 FROM "UserProfiles" WHERE "supId" = NEW.user_id
+    ) INTO user_exists;
+    
+    -- If user doesn't exist in UserProfiles, try to create it from auth.users
+    IF NOT user_exists THEN
+        BEGIN
+            INSERT INTO "UserProfiles" ("supId", "supEmail", "name", "email", "updatedAt")
```

Comment:
I think we should create the UserProfile here as well with similar SQL to https://github.com/wanderwallet/embed-api/blob/23cd5a47e6f4aa28e44dd32a9c6048dfed2a43ea/prisma/migrations/20250221160426_user_trigger/migration.sql#L16